python/tskit/tables.py

```diff
@@ -582,6 +582,20 @@ def append(self, row):
             }
         )
 
+    def replace_with(self, other):
+        """
+        Entirely replace the contents of this table with another
+        """
+        if type(self) != type(other):
+            raise TypeError("replacement column is not of the same type")
```

Comment:
Do you mean table here instead of column?

Reply:
Good spot, thanks!